src/main/java/org/truffleruby/cext/CExtNodes.java

```diff
@@ -2071,4 +2071,35 @@ RubyArray zlibGetCRCTable() {
         }
     }
 
+    @CoreMethod(names = "rb_frame_method_and_id", onSingleton = true, required = 2)
+    public abstract static class FrameMethodAndId extends CoreMethodArrayArgumentsNode {
+
+        @Specialization
+        boolean frameMethodAndId(Object frameMethod, Object frameId) {
+            final Frame callingMethodFrame = findCallingMethodFrame();
+            frameMethod = RubyArguments.getMethod(callingMethodFrame);
+            frameId = System.identityHashCode(RubyArguments.tryGetSelf(callingMethodFrame));
```

Comment:
These assignments have no effect, they only assign the local Java variable, not the C pointer.
You need to return something which contains both (e.g. a Method Ruby object or a Ruby Array of two elements).